src/solve.jl

```diff
@@ -75,7 +75,7 @@ function init(prob::AbstractDDEProblem{uType,tType,lType,isinplace}, alg::algTyp
                                                        ode_prob,
                                                        OrdinaryDiffEq.alg_order(alg)))
     end
-    integrator.dt = dt
+    integrator.dt = zero(dt)
```

Comment:
>But it is not implemented yet, is it? As far as I see, all calls of current_interpolant and current_interpolant! (and also of current_extrapolant and current_extrapolant!) will calculate a Θ that is not a number for dt=0 - which will always lead to interpolations/extrapolations that are not a number, I guess.

But almost all of the interpolations have `u0 + dt* ...`, so it'll "accidentally" work out as a constant extrapolation. I think just the SSPRK methods don't. But I don't think we're actually hitting this case so it probably doesn't come up. Probably worthwhile to address in another PR if it does, with the SSPRK methods/interpolants as the test case which doesn't have the accidental workaround.